crates/sui-core/src/validator_tx_finalizer.rs

```diff
@@ -0,0 +1,426 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::authority::AuthorityState;
+use crate::authority_aggregator::AuthorityAggregator;
+use crate::authority_client::AuthorityAPI;
+use std::sync::{Arc, Weak};
+use std::time::Duration;
+use sui_types::transaction::VerifiedSignedTransaction;
+use tokio_util::sync::CancellationToken;
+use tracing::{debug, error};
+
+/// Only wake up the transaction finalization task for a given transaction
+/// after 15 mins of seeing it. This gives plenty of time for the transaction
+/// to become final in the normal way.
+const TX_FINALIZATION_DELAY: Duration = Duration::from_secs(15 * 60);
```

Comment:
Why wait so long? I would have expected maybe around 30 seconds

Reply:
Did you mean to ask about the TIMEOUT? This one is to only process a transaction after 15 mins seeing it.

Reply:
No, I meant to ask about this one. 99% of the time (maybe 99.99%), if we wait 30 seconds, we will just immediately exit at the `is_tx_already_executed` check, without doing any work. So what is waiting so long buying us?